## First Step With Python and Jupyter Notebook

> Content

1. Performing arithmetic operations using Python
2. Solving multi-step problems using variables
3. Evaluating conditions with logical operators
4. Adding text styles using Markdon

### 1. Performing Arithmetic Operations Using python

![](https://i.imgur.com/gvSnw4A.png)

#### Python As Calculator

In [19]:
# Addition

2+3+9+7

21

In [20]:
# Substaction

23.5-.2

23.3

In [21]:
# Multiplication

100*10

1000

In [23]:
# Division

10/3

3.3333333333333335

In [24]:
# Only Integer Part Of Division

10//3

3

In [25]:
# Only Reminder

10%3

1

In [27]:
# Power Form

5**3

125

In [28]:
# Combinations

2+2-2*2/2

2.0

In [30]:
(2+2)-(2*2)/(2*2)

3.0

In [32]:
((2+5)*(17-3)/4**3)

1.53125

#### Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |

### 2. Solving Multi-Level Problems Using Variables

##### Important definition and commands used in this chapter

> **Variables:-** Variables are used to store the informations. We can think of variables as container for storing data.    
*Example:-*  x, Men, Women, Total_amount etc...  
 *(Note:- Usually left-side of  "="  operator is called the variable name)* 



> **value:-** The data stored within the variable is  called value.  
*Example:-*  20, 45312, 35678, Total_Population-Men-Women etc...  
 *(Note:- Usually Right-side of  "="  operator is called the variable name)*

> **Functions:-** A function is a reusable set of instructions. It takes one or more inputs to perform certain operations, and often returns an output.  
*Note:-*  Python provides many in-built functions like `print` and also allow us to define our own functions.

> **`print`** :- `print` is a function to print or display the quoted information. It takes one or more inputs, which can be text (within quotes, e.g.- `'Hello World'`, `"This is some text"`), numbers, variables, mathematical expressions etc...

> **Comments:-** Comments and blank lines are ignored during execution, but they are useful for providing information to humans (including yourself) about what the code does. Comments can be `inline`- `#_ _ _` (at the end of same code), on a separate line, or even  span `multiple lines`- `'''_ _ _'''`.

> *' # '* :- Use for line comment    
*Example:-* # Solution Of Question no.1

>  '''_____''' :- Use for Multi-Line comments or text  
*Example:-*   
'''
Question No. 1- The population of a town is 198568. Out of them 45312 are men and 35678 are women. 
Find the number of children in the town.
'''

> Solution Of Given Problem Sets:
* https://www.math-only-math.com/worksheet-on-word-problems-on-four-operations.html .

In [34]:
'''
Question No. 1- The population of a town is 198568. Out of them 45312 are men and 35678 are women. 
Find the number of children in the town.

'''
# Solution Of Question no.1

Total_Population = 198568
Men = 45312
Women = 35678
Children = Total_Population-Men-Women
print("The Number Of Children In The Town: ", Children)

# Note:- Here we use print function to print the quoted text with the required outcomes


The Number Of Children In The Town:  117578


In [36]:
''' 
Question No.2- A shopkeeper has 2425 boxes of 24 pencils each. How many pencils do all the boxes have in all?

'''
# Solution Of Question No.2
Boxes = 2425
Pencils_In_One_Box = 24
Total_Pencils = Boxes*Pencils_In_One_Box
print("Total Pencils in all the boxes: ", Total_Pencils)

Total Pencils in all the boxes:  58200


In [40]:
''' 
Question No.3- Linda bought a coat for $2265 and a saree for $2150. She gave $5000 to the shopkeeper. 
How much money did the shopkeeper return to her? 

'''
# Solution Of Question No.3

Cost_of_a_coat = 2265
Cost_of_a_saree = 2150
Amount_given = 5000
Return_money = Amount_given - Cost_of_a_coat - Cost_of_a_saree
print("Shopkeeper returned the money to Linda ($):", Return_money)

Shopkeeper returned the money to Linda ($): 585


In [45]:
''' 
Question No.4- The cost of 21 TV sets is $95844. Find the cost of one TV set.

'''
# Solution Of Question no.4

Number_of_tv = 21
Total_cost_of_tv = 95844
Cost_Of_One_tv = Total_cost_of_tv / Number_of_tv
print("Cost Of one TV ($):",Cost_Of_One_tv)

Cost Of one TV ($): 4564.0


In [46]:
'''
Question No.5- A factory produces 24532 bulbs in a month. What is its annual production

'''
# Solution Of Question No.5

Numbers_of_bulbs_in_a_month = 24532
Annual_production_of_bulbs = 12*Numbers_of_bulbs_in_a_month
print("Annual production of bulbs in factory: ",Annual_production_of_bulbs)

Annual production of bulbs in factory:  294384


In [51]:
''' 
Question No.6- There are 145968 bags of sugar, 236487 bags of wheat and some bags of rice in a godown. 
It the total number of bags in the godown is 450000, find the number of bags of rice.

'''
# Solution Of Question No.6

Sugar_bags = 145968
Wheat_bags = 236487
Total_number_of_bags = 450000
Rice_bags= Total_number_of_bags - Sugar_bags - Wheat_bags
print("Total number of Rice bags:",Rice_bags )

Total number of Rice bags: 67545


In [52]:
''' 
Question No.7- A factory manufactured 483685 toys in three weeks. The production in first week was 146345 toys and 
in second week 138152 toys. Find the production in the third week?

'''
# Solution Of Question No.7

Toys_in_three_weeks = 483685
Toys_in_first_week = 146345
Toys_in_second_week = 138152
Toys_in_third_week = Toys_in_three_weeks - Toys_in_first_week -Toys_in_second_week
print("Production of toys in third week:", Toys_in_third_week)

Production of toys in third week: 199188


In [53]:
'''
Question No.8- The cost of a sofa set is $9372. How much will 124 such sofa sets cost?

'''
# Solution Of Question No.8

Cost_of_a_sofaset = 9372
Total_cost = 124*Cost_of_a_sofaset
print("Total cost of 124 Sofa-Sets ($):",Total_cost)

Total cost of 124 Sofa-Sets ($): 1162128


In [55]:
''' 
Question No.9-  There are 86 rooms in a school. 4356 students study there. Equal number of students sits in each room?

'''
# Solution Of Question No.9

total_rooms_in_school = 86
total_student = 4356
equal_number_of_students_in_each_rooms = total_student//total_rooms_in_school
print("The equal number of sudents sits in each room:",equal_number_of_students_in_each_rooms)

The equal number of sudents sits in each room: 50


In [61]:
''' 
Question No.10- 1575 students of a school want to go Agra by bus. If one bus can carry 75 students, 
how many buses are required to carry all the students? 

'''
# Solution Of Question No.10

total_students = 1575
one_bus_carry_students = 75
total_bus_required = total_students/one_bus_carry_students
print("Total number of buses required: ",int(total_bus_required))

Total number of buses required:  21


In [64]:
''' 
Question No.11- The cost of a radio set is 1475. What is the cost of 35 such radio sets?

'''
# Solution Of Question No.11

cost_of_a_radio_set = 1475
cost_of_35_radio_sets = 35*cost_of_a_radio_set
print("Cost of 35 radio sets ($):",cost_of_35_radio_sets)

Cost of 35 radio sets ($): 51625


In [65]:
''' 
Qurestion No.12- In an election, 52496 people voted for Ron, 44929 people for Jhon and 36824 people for Mike in a town. 
If everyone voted in the town, what is the total number of voters? 

'''
# Solution Of Question No.12

Voted_for_Ron = 52496
Voted_for_Jhon = 44929
Voted_for_Mike = 36824
Total_voters = Voted_for_Ron + Voted_for_Jhon + Voted_for_Mike
print("Total numbers of voters in Town:",Total_voters)

Total numbers of voters in Town: 134249


In [88]:
'''
Question No.13-  Maria bought 96 toys priced equally for $12960. The amount of $1015 is still left with her. 
Find the cost of each toy and the amount she had.

'''
# Solution Of Question No.13

Total_toys = 96
Total_equally_priced = 12960
Remaining_amount = 1015
Cost_of_each_toy = Total_equally_priced/Total_toys
Total_amount = Total_equally_priced + Remaining_amount
print("Cost of each toy ($):",Cost_of_each_toy ,'\n' "Total amount Maria had ($):",Total_amount)


Cost of each toy ($): 135.0 
Total amount Maria had ($): 13975


In [96]:
''' 
Exercise:- A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. 
So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, 
what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

'''
# Solution

Flying_cost = 5000
Number_of_people_signed = 29
Cost_of_per_ticket = 200
Revenue_of_company = Number_of_people_signed*Cost_of_per_ticket
Profit_made_by_the_company = Revenue_of_company - Flying_cost
print("Profit made by the company ($):", Profit_made_by_the_company)

Profit made by the company ($): 800


### 3. Evaluating Conditions Using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers and variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [1]:
Odd_number = 1
Even_number=2
Even_odd=0


In [2]:
# Equality Check- True
Odd_number == 1

True

In [3]:
# Equality check- False
Odd_number == Even_number

False

In [4]:
# Not equal check- True
Odd_number !=2  

True

In [5]:
# Not equal check- False
Odd_number != 1

False

In [6]:
# Grater then check- True
Even_number>Odd_number

True

In [7]:
# Grater then check- False
Odd_number>1

False

In [8]:
# less than check- True
1<Even_number

True

In [9]:
# Less than check- False
Odd_number<1

False

In [10]:
# Grater than or equal check- True
Odd_number>=1

True

In [11]:
# Grater than or equal check- False
Even_number>=3

False

In [13]:
# Less than or equal check- True
1<=Odd_number

True

In [14]:
# Less than or equal check- False
Even_number<=1

False

##### **Note ;-**  Just like arithmetic operations, the result of a comparison operation can also be stored in a variable.

In [15]:
Cost_of_per_ticket=200
is_travelling_expensive=Cost_of_per_ticket>=250
print("Is travelling expensive? - ", is_travelling_expensive)

Is travelling expensive? -  False


### 4. Combining Conditions With Logical Operators

The logical operators `and`, `or`, and `not` operate upon conditions and `True` and `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on single condition.  


#### And operator :-  
The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False` see the following table-

|   `a`   |    `b`   | `a and b` |
|- - - - -| - - - - -| - - - - - |
|  `True  |  `True`  |   `True`  |
| `True`  |  `False` |  `False`  |
|  `False`|  `True`  |  `False`  |
|  `False`|  `False` |  `False`  |

In [17]:
Odd_number

1

In [18]:
Even_number

2

In [21]:
Even_odd


0

In [24]:
# True `and` True = True
Even_odd < Odd_number and Odd_number <= Even_number

True

In [26]:
# True `and` False = False
Odd_number>=1 and Odd_number>=Even_number


False

In [27]:
# False `and` True = False
Even_number<1 and Even_number>=1

False

In [28]:
# False `and` False = False
Even_number<= Odd_number and Odd_number<=Even_Odd

False

#### Or Operator ;-  
The `or` operator returns `True` if at least one of the conditions evaluate to `True`. It returns `False` only if both conditions are `False`.

|   `a`   |   `b`   |  `a or b`  |
| - - - - |- - - - -|- - - - - - |
| `True`  | `True`  |   `True`   |
| `True`  | `False` |  `True`    |
| `False  |  `True` |  `True`    |
| `False` | `False` | `False`    |

In [29]:
Odd_number= 5
Even_number= 10
Odd_Even= 0

In [30]:
# True `or` True = True
Odd_number<Even_number or Odd_Even<Even_number

True

In [31]:
# True `or` False = True
Even_number<=100 or Odd_number>=Even_number

True

In [32]:
# False `or` True = True
Odd_Even>=Odd_number or Odd_number>Odd_Even

True

In [33]:
# False `or` False = False
Even_number<=Odd_number or Odd_Even>=Even_number

False

#### Not Operator :-  
The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

|  `a`  | `not a` |
|- - - -|- - - - -|
|`True` | `False` |
|`False`| `True`  |

In [34]:
Odd_number=3
Even_number=6
Odd_Even=0

In [35]:
# True `not` True = False 
not Odd_number==3

False

In [36]:
# False `not` False = True
not Odd_number==6

True

In [37]:
# Direct Operator- `not` True = False
not True

False

In [38]:
# Direct Operator- `not` False = True

not False

True

#### Combination Of Logical Operators

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [39]:
(5<10 and 10>1) or not(Odd_number>=Odd_Even and Odd_Even>Even_number)

True

In [40]:
not(True and 0>1) and (False or 1<0)

False

In [44]:
not (True and False) and (not False or True)

True

### 5. Adding Text Style using Markdown

Adding explanation using text cells (like this one) is a great way to make your notebook informative for the readers. It also useful if you need to refer back to it in the future. You can *Double Click* on the text cell within *Jupyter* to edit it. In the edit mode, you'll notice that the text looks slightly different (for instance, the heading has `##` prefix). This text is written using `Markdown`, a simple way to add styles to your text. Execute this cell to see the output without the special characters. You can switch back and forth between the source and the output to apply a specific style.

**1. For instance, you can use one or more `#` characters at the stat of line to create headers of different sizes:**

# Header 1

## Header 2

### Header 3

#### Header 4

**2. To create a bulleted `.` or numbered list `1,2,3...`, simply start a line with `*` or `1.` .**

A bulleted list:

* Item 1
* Item 2
* Item 3

A numbered list:

1. Item A
2. Item B
3. Item C

**3. You can make some text bold using `**`**

**Example :-  Some bold texts.**

**4. You can make texts italic using `*`**

*Example :-  Some italic text.*

**5. You can also create links using this sample**

e.g.:- 
    [Brainwork - The Monkey Mind](https://www.youtube.com/watch?v=blwB2DLnjEs)

**6. You can easily embedded *Image* too**

![](http://codeskulptor-demos.commondatastorage.googleapis.com/GalaxyInvaders/back07.jpg)

**7. Another really nice feature of *Markdown* is the ability to include blocks of code. Note that code blocks inside *Markdown* cells cannot be executed.**

```
# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# Display the result
print("The grocery store makes a total profit of $", total_profit)

```

You can learn the full syntax of Markdown here: https://learnxinyminutes.com/docs/markdown/

### Save and Upload your notebook

Whether you're running Jupyter notebook online or on computer, it's essential to save your work from time to time. You can continue working on a saved notebook later or share it with friends and colleagues to let them execute your code. Jovion Offers an easy way of saving and sharing your Jupyter notebooks online with following steps-

**Step-1**

You need to install the Jovian python library if it isn't already installed

In [51]:
!pip install jovian --upgrade --quiet

**Step-2**

The library needs to be imported

In [53]:
import jovian

**Step-3**

Finally, you can run jovian.commit to capture and upload snapshot of the notebook.

In [ ]:
jovian.commit(project='Chapter-1')

<IPython.core.display.Javascript object>

**Step-4**

The first time you run `jovian.commit`, you'll be asked to provide an _API Key_ to securely upload the notebook to your Jovian account. You can get the API key from your [Jovian profile page](https://jovian.ai) after logging in / signing up.

<img src="https://i.imgur.com/taLLUVd.png" width="480">

`jovian.commit` uploads the notebook to your Jovian account, captures the Python environment, and creates a shareable link for your notebook, as shown above. You can use this link to share your work and let anyone (including you) run your notebooks and reproduce your work. Jovian also includes a powerful commenting interface, so you can discuss & comment on specific parts of your notebook:

![https://jovian.ai/docs/user-guide/upload.html](https://i.imgur.com/kxx3pqM.png)

You can do a lot more with the `jovian` Python library. Visit the documentation site to learn more: https://jovian.ai/docs/index.html

## Further Reading and References

Following are some resources where you can learn about more arithmetic, conditional and logical operations in Python:

* Python Tutorial at W3Schools: https://www.w3schools.com/python/
* Practical Python Programming: https://dabeaz-course.github.io/practical-python/Notes/Contents.html
* Python official documentation: https://docs.python.org/3/tutorial/index.html

Now that you have taken your first steps with Python, you are ready to move on to the next tutorial: ["A Quick Tour of Variables and Data Types in Python"](https://jovian.ml/aakashns/python-variables-and-data-types).

## Question and Answer for Revision

Try answering the following questions to test your understanding of the topics covered in this notebook:

**1. What is a Jupyter notebook?**   
**Ans :-**  Jupyter notebook is a text editor for python. It fulfills all the needs of Python coders.

**2. How do you add a new code cell below an existing cell?**  
**Ans :-** `Esc+B`  or click on `Insert` then choose `insert cell below`

**3. How do you add a new Markdown cell below an existing cell?**  
**Ans :-** `Esc+B` and `Esc+M` or click on `insert` then choose `insert heading below`  

**4. How do you convert a code cell to a Markdown cell or vice versa?**  
**Ans :-** Code cell to Markdown- `Esc+M` & Markdown to code cell `Esc+C`  

**5. How do you execute a code cell within Jupyter?**  
**Ans :-** `Shift + Enter` or click on `Run`  

**6. What the different arithmetic operations supported in Python?**  
**Ans :-** Arithmetic operations - `+,-,*,/,%`(reminder),`//`(interger part), `**`(power form)

**7. How do you perform arithmetic operations using Python?**  
**Ans :-** We directlly perform the arithmetic operations in code cells.  

**8. What is the difference between the `/` and the `//` operators?**  
**Ans :-** `/`- used for division & `//`- used for only integer part of a division.  

**9. What is the difference between the `*` and the `**` operators?**  
**Ans :-**  `*` -used for multiplications & `**` -used for exponent form or power form  

**10. What is the order of precedence for arithmetic operators in Python?**  
**Ans :-** Python Operators Precedence Rule – **PEMDAS**  
You might have heard about the BODMAS rule in your school’s mathematics class. Python also uses a similar type of rule known as PEMDAS.

**P – Parentheses  
E – Exponentiation  
M – Multiplication  
D – Division  
A – Addition  
S – Subtraction**

The precedence of operators is listed from High to low. To remember the abbreviations, we have a funny mnemonic  *“Please Excuse My Dear Aunt Sally”*.  

**11. How do you specify the order in which arithmetic operations are performed in an expression involving multiple operators?**  
**Ans :-** Using **PEMDAS** method.  

**12. How do you solve a multi-step arithmetic word problem using Python?**  
**Ans :-** We use variables to solve the multi-level arithmetic word problem in Python.  

**13. What are variables? Why are they useful?**
**Ans :-** variables are like container in which we store data. It's use because we can use it multiple times once it defined.  

**14. How do you create a variable in Python?**  
**Ans :-** We create variables using `alphabets` (Upper & Lower both case), `_`, and `numbers` in a fixed manner.  

**15. What is the assignment operator in Python?**  
**Ans :-** `=` is said to be assignment operator because it assign the value to the variables.  

**16. What are the rules for naming a variable in Python?**  
**Ans :-** We create variables using `alphabets` (Upper & Lower both case), `_`, and `numbers` in required combinations. Variables are case sensitive so be careful. Don't use `Special characters, Spaces, predefined variables` etc..  

**17. How do you view the value of a variable?**  
**Ans :-** Just type the name of variable or use `print` function.  

**18. How do you store the result of an arithmetic expression in a variable?**  
**Ans :-** We just define a variable name with assignment operator to store the value of an arithmetic expression.  

**19. What happens if you try to access a variable that has not been defined?**  
**Ans :-** It'll show the `name error` ie.variable is not define.  

**20. How do you display messages in Python?**  
**Ans :-** using Markdown form or comments command (`#` or `'''_ _ _'''`).  

**21. What type of inputs can the print function accept?** 
**Ans :-** The `print` function takes only text, variables,and expressions types inputs.  

**22. What are code comments? How are they useful?**  
**Ans :-** Code comments are define by single line- `#` or multi-line- `'''_ _ _'''`. It's useful for the readers and yourself to understand the code.  

**23. What are the different ways of creating comments in Python code?**  
**Ans :-** There are two way to create comments. First- single line comment using `#` before the comment text and Second- multi-line comment using `'''` before and after the comment texts.  

**24. What are the different comparison operations supported in Python?**  
**Ans :-** The different comparison operators in Python are - `<,>,==,<=,>=`.  

**25. What is the result of a comparison operation?**
**Ans :-** There is only two results either `True` or `False`.  

**26. What is the difference between `=` and `==` in Python?**  
**Ans :-** `=` is an assignment operator (to assign a value) where as `==` is a comparison operator (to compare the value).  

**27. What are the logical operators supported in Python?**  
**Ans :-** Mainly three logical operators exits in python - `and, or, not`.  

**28. What is the difference between the `and` and `or` operators?**  
**Ans :-** when both the conditions are True then`and`operator gives- `True` Otherwise it gives `False`.  
When any of one condition is `True` then `or` gives -`True`. It gives `False` iff both condition are `False`.  

**29. Can you use comparison and logical operators in the same expression?**  
**Ans :-** yes, we can use as many different operations as we want in a prescribed manner.  

**30. What is the purpose of using parentheses in arithmetic or logical expressions?**  
**Ans :-** Parentheses are used to define the sequence of operations as we want to perform. It's very useful for long expressions.  

**31. What is Markdown? Why is it useful?**  
**Ans :-** Markdown is a environment in which we write, edit and describe the text formate or code formate. It's very useful for coders to understand the program .  

**32. How do you create headings of different sizes using Markdown?**  
**Ans :-** We use `# + Space`to create the heading. We increase the number of `#'s` to decrease th size of heading.  

**33. How do you create bulleted and numbered lists using Markdown?**  
**Ans :-** We use `*` to create bulleted list and `1., 2., 3.` to create numbered lists.  

**34. How do you create bold or italic text using Markdown?**  
**Ans :-** We use `**` for **bold Text** whereas `*` for *italic text*.  

**35. How do you include links & images within Markdown cells?**  
**Ans :-** To include link we use- `[link_name](link)`.  
To include images we use- `![](link_of_image)`.  

**36. How do you include code blocks within Markdown cells?**  
**Ans :-** We just use `''' and #` to include code blocks but it doesn't executed.  

**37. Is it possible to execute the code blocks within Markdown cells?**  
**Ans :-** No.  

**38. How do you upload and share your Jupyter notebook online using Jovian?**  
**Ans :-** First we install the Jovian python library - then we import the Jovian Library- then we run the command `jovion.commit(name_of_notebool)` to upload or share - then we need to enter the API's key for the first time after that no need.  

**39. What is the purpose of the API key requested by jovian.commit ? Where can you find the API key?**  
**Ans :-** The purpose of API key is to secure and save the notebook at `jovian.ai`. We can find the API key click on the link we get when we use the `jovian.commit` command ie. in aakashns profile.  

**40. Where can you learn about arithmetic, conditional and logical operations in Python?**  
**Ans :-** There are many but mainly-  
* Python tutorial of W3school
* Python documentation
* practical python programming